src/scrapers/base-scraper-with-browser.ts

```diff
@@ -169,6 +131,57 @@ class BaseScraperWithBrowser<TCredentials extends ScraperCredentials> extends Ba
     });
   }
 
+  private async initializePage() {
+    debug('initialize browser page');
+    if ('browserContext' in this.options) {
+      debug('Using the browser context provided in options');
+      return this.options.browserContext.newPage();
+    }
```

Comment:
The common way to do this condition is `if (this.options.browserContext)`

Reply:
Since `ScraperBrowserOptions` is now a union type and `this.options.browserContext` will emit the following error:
```
Property 'browserContext' does not exist on type 'ScraperBrowserOptions'.
  Property 'browserContext' does not exist on type 'ExternalBrowserOptions'.
```

Only after the `'browserContext' in this.options` check, TS can narrow the type of `this.options` to `ExternalBrowserContextOptions`